include/seqan3/utility/char_operations/predicate_detail.hpp

```diff
@@ -194,7 +194,10 @@ struct char_predicate_base
     constexpr bool operator()(value_t const val) const noexcept
         requires (sizeof(value_t) != 1)
     {
-        using char_trait = std::char_traits<value_t>;
+        // std::char_traits is only defined for char types. libc++ deprecates other specialisations in llvm-17, and
+        // removes them in llvm-18. The int specialisation is needed for seqan3::is_eof(EOF).
+        using char_traits_value_t = std::conditional_t<std::same_as<value_t, int>, char, value_t>;
```

Comment:
help me out. 
* If `value_t == int` then `char_traits_value_t == char`
* If `value_t == char` then `char_traits_value_t == char`
* If `value_t == other` then `char_traits_value_t == other`
correct ?

So in case of `other` and `other not a char type`, `std::char_traits` is still not defined in libc++?
I don't really understand your fix

Reply:
The error is

```
predicate_detail.hpp:200:33: error: 'char_traits<int>' is deprecated: char_traits<T> for T not equal to char, wchar_t, char8_t, char16_t or char32_t is non-standard and is provided for a temporary period. It will be removed in LLVM 18, so please migrate off of it. [-Werror,-Wdeprecated-declarations]
  200 |         using char_trait = std::char_traits<value_t>;
```

https://en.cppreference.com/w/cpp/string/char_traits

> Several explicit specializations are provided for the standard character types (see below), other specializations are not required to satisfy the requirements of CharTraits.

So, only those lists in specialisations (`char`, `wchar_t`, `char8_t`, `char16_t` or `char32_t`) are standard-conform. The compilers just didn't warn for the other overloads, or maybe even just provided them for integer types.

We could also

> The user can provide a specialization for any custom character types.

Or even only do an overload for `EOF` only?

Reply:
See `[REVIEW] char_predicate alternative` commit

Reply:
lets go over this in webex